src/indexer/segment_writer.rs

```diff
@@ -242,7 +242,9 @@ impl SegmentWriter {
                         )
                     };
 
-                    self.fieldnorms_writer.record(doc_id, field, num_tokens);
+                    if field_entry.has_fieldnorms() {
+                        self.fieldnorms_writer.record(doc_id, field, num_tokens);
```

Comment:
I think fieldnorms_writer should be optional and the check simply if the writer exists

Reply:
Agreed